pkg/source/dynamic.go

```diff
@@ -8,14 +8,13 @@ import (
 	"k8s.io/client-go/util/workqueue"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
-	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 // Stoppable is a stoppable source
 type Stoppable interface {
 	source.Source
-	inject.Stoppable
+	InjectStopChannel(<-chan struct{}) error
```

Comment:
Are we sure we should just be doing this update? It looks like the general path to this is no longer really supported upstream so I wonder if we need to do a larger change here.

Reply:
Seems like we don't have any reference to using the `Stoppable` in our repos: https://github.com/search?q=org%3Asolo-io%20Stoppable&type=code, and it is only used in this function signature: https://github.com/solo-io/skv2/blob/k8s-1.27-upgrade/pkg/source/dynamic.go#L99

Would it be appropriate to some kind of fail/panic to that `Add` function as an interim approach to 1) make sure none of the code we're upgrading as part of the k8s 1.27 is using it and 2) make sure if someone down the line tries to use it they get quick feedback instead of potentially having to debug a mysterious failure?

Reply:
change added below in `Add`. Nothing downstream seems to be having issues.